src/SciMLOperators.jl

```diff
@@ -29,70 +22,35 @@ abstract type AbstractSciMLOperator{T} end
 """
 $(TYPEDEF)
 """
-abstract type AbstractDiffEqOperator{T} <: AbstractSciMLOperator{T} end
+abstract type AbstractSciMLLinearOperator{T} <: AbstractSciMLOperator{T} end
 
 """
 $(TYPEDEF)
 """
-abstract type AbstractDiffEqLinearOperator{T} <: AbstractDiffEqOperator{T} end
+abstract type AbstractSciMLCompositeOperator{T} <: AbstractSciMLOperator{T} end
 
 """
 $(TYPEDEF)
 """
-abstract type AbstractDiffEqCompositeOperator{T} <: AbstractDiffEqLinearOperator{T} end
-
-"""
-$(TYPEDEF)
-"""
-abstract type AbstractMatrixFreeOperator{T} <: AbstractDiffEqLinearOperator{T} end
+abstract type AbstractMatrixFreeOperator{T} <: AbstractSciMLOperator{T} end
 
+include("traits.jl")
+include("basic.jl")
+include("sciml.jl")
 include("interface.jl")
-include("operators/basic_operators.jl")
-include("operators/diffeq_operator.jl")
-include("operators/matrixfree_operators.jl")
-#include("operators/composite_operators.jl")
-include("operators/common_defaults.jl")
-
-# Define a helper function `sparse1` that handles
-# `DiffEqArrayOperator` and `ScaledDiffEqOperator`.
-# We should define `sparse` for these types in `SciMLBase` instead,
-# but that package doesn't know anything about sparse arrays yet, so
-# we'll introduce a temporary work-around here.
-_sparse(L) = sparse(L)
-_sparse(L::DiffEqArrayOperator) = _sparse(L.A)
-_sparse(L::ScaledDiffEqOperator) = L.λ * _sparse(L.L)
-
-# (u,p,t) and (du,u,p,t) interface
-for T in (
-          DiffEqIdentity,
-          DiffEqNullOperator,
-          DiffEqScalar,
-          ScaledDiffEqOperator,
-          AddedDiffEqOperator,
-
-          DiffEqArrayOperator,
-          FactorizedDiffEqArrayOperator,
 
-#         DiffEqOperatorCombination,
-#         DiffEqOperatorComposition,
-         )
-
-    (L::T)(u, p, t) = (update_coefficients!(L, u, p, t); L * u)
-    (L::T)(du, u, p, t) = (update_coefficients!(L, u, p, t); mul!(du, L, u))
-end
-
-export DiffEqScalar,
-       DiffEqArrayOperator,
-       FactorizedDiffEqArrayOperator,
-       AffineDiffEqOperator,
-       MatrixFreeOperator
+export SciMLScalar,
+       SciMLMatrixOperator,
+       SciMLFactorizedOperator,
+       AffineSciMLOperator,
+       SciMLFunctionOperator
```

Comment:
should be ok so long as we don't export these

Reply:
It would only export if someone `using SciMLOperators`, which would be a pretty deliberate thing that only exports the operator types. I don't think any of the solver libraries will export the basic operators, so this should be fine I think.

Reply:
Also, I don't think these names would clash with anything. `Scalar` and `Identity` would be bad ideas to export though 😅

Reply:
done